internal/graph/check.go

```diff
@@ -245,33 +243,36 @@ func (cc *ConcurrentChecker) checkDirect(ctx context.Context, crc currentRequest
 			return childResult
 		}
 
-		return mapResourceIds(childResult, dd.resourceType, resourceIDsBySubjectID)
+		return mapFoundResources(childResult, dd.resourceType, relationshipsBySubjectONR)
 	}, cc.concurrencyLimit)
 
-	return combineResultWithFoundResourceIds(result, foundResourceIds)
+	return combineResultWithFoundResources(result, foundResources)
 }
 
-func mapResourceIds(result CheckResult, resourceType *core.RelationReference, resourceIDsBySubjectID *util.MultiMap[string, string]) CheckResult {
+func mapFoundResources(result CheckResult, resourceType *core.RelationReference, relationshipsBySubjectONR *util.MultiMap[string, *core.RelationTuple]) CheckResult {
 	// Map any resources found to the parent resource IDs.
-	mappedResourceIds := []string{}
+	membershipSet := NewMembershipSet()
 	for foundResourceID, result := range result.Resp.ResultsByResourceId {
-		if result.Membership != v1.DispatchCheckResponse_MEMBER {
+		if result.Membership != v1.ResourceCheckResult_MEMBER {
```

Comment:
Wouldn't the dispatched response come back with potentially caveated results? Wouldn't this be discarding those?

Reply:
Ah, I see, the caveat is being added here, because the tuple was loaded in this dispatching node.

Reply:
Yes, good catch.

Reply:
Would it make sense to add some tests that exercise dispatching caveated tuples? Perhaps mocking the response. It felt like this PR lacked testing around the new conditions. I run checks_test.go and the debugger didn't stop in any of the paths affected by the PR.

Reply:
It explicitly lacks those tests: this PR is to ensure that the *existing* check behavior works with the new membership set. All other considerations are out of scope until the *next* PR, which will be exercising the caveats once we have a test harness built for it (which is what I will be doing next)

Reply:
Alright, fair, let's do that!